src/saltext/vmware/states/esxi.py

```diff
@@ -1,3 +1,260 @@
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Remove local users_by_host on matching ESXi hosts.
+
+    name
+        User to create on matching ESXi hosts. (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: yaml
+
+    Remove User:
+      vmware_esxi.user_absent:
+        - name: local_user
+
+    """
+    log.debug("Running vmware_esxi.user_absent")
+    ret = {"name": name, "result": None, "comment": "", "changes": {}}
+    delete = 0
+    failed_hosts = []
+    diff = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    users_by_host = __salt__["vmware_esxi.get_user"](
+        user_name=name,
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    hosts = __salt__["vmware_esxi.list_hosts"](
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    for host in hosts:
+        if host in users_by_host:
+            delete += 1
+            diff[host] = {name: True}
+        else:
+            diff[host] = {name: False}
+
+    for host in diff.copy():
+        if __opts__["test"]:
+            if not ret["comment"]:
+                ret["comment"] = "User {} will be deleted on {} host(s).".format(name, delete)
+                ret["result"] = None
+        elif diff[host][name]:
+            try:
+                __salt__["vmware_esxi.remove_user"](
+                    user_name=name,
+                    datacenter_name=datacenter_name,
+                    cluster_name=cluster_name,
+                    host_name=host,
+                    service_instance=service_instance,
+                )
+            except salt.exceptions.SaltException as exc:
+                delete -= 1
+                failed_hosts.append(host)
+                diff.pop(host)
+                ret[
+                    "comment"
+                ] = "User {} removed on {} host(s), failed on {} host(s). List of failed host(s) - {}. Sample Error: {}".format(
+                    name, delete, len(failed_hosts), ",".join(failed_hosts), exc
+                )
+                ret["changes"] = diff
+                ret["result"] = False
+            if not ret["comment"]:
+                ret["comment"] = "User {} removed on {} host(s).".format(name, delete)
+                ret["changes"] = diff
+                ret["result"] = True
+        else:
+            if not ret["comment"]:
```

Comment:
I think this should happen at the very end, outside of the loop.  Right now, this message is only emitted if the first host in the list doesn't have the user, and it masks any info about other hosts.

If there were hosts, but there is no comment when we go to return, the user didn't exist on any host.

Reply:
ok will do

Reply:
Thanks for your patience, Sud :)